File: Python/two_sum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            diff= target-nums[i]
            if diff in nums[i+1:]:
                ind= nums[i+1:].index(diff)
                return [i,ind+i+1]

File: Go/two_sum.go
package Go

// Two Sum problem from Leetcode

// https://leetcode.com/problems/two-sum/

func twoSum(nums []int, target int) []int {
	history := map[int]int{}
	result := []int{}

	for i, n := range nums {

		temp := target - n

		if j, ok := history[temp]; ok {
			result = append(result, j)
			result = append(result, i)

			return result
		}

		history[n] = i
	}

	return result
}
